agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java

```diff
@@ -446,23 +459,48 @@ private static String getWebsiteSiteNameEnvVar() {
     return value;
   }
 
+  public static String overlayWithSysPropEnvVar(
+      String systemPropertyName, String envVarName, String defaultValue) {
+    String value = getSystemProperty(systemPropertyName);
```

Comment:
system properties have a higher precedence over env vars?  should it be the other way around? env vars are user defined properties and i think it has a higher precedence over the default system properties, right? we do want to respect end user's choice?

Reply:
system properties are also provided by end user

this precedence ordering is same as Spring (and OTel also follows same):

https://docs.spring.io/spring-boot/docs/2.5.2/reference/html/features.html#features.external-config